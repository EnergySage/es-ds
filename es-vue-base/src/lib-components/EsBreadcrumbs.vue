<template>
    <b-breadcrumb
        v-bind="$attrs"
        v-on="$listeners" />
</template>
<script>
import { BBreadcrumb } from 'bootstrap-vue';

export default {
    name: 'EsBreadcrumb',
    components: {
        BBreadcrumb,
    },
};
</script>

<style lang="scss" scoped>
@import '~@energysage/es-bs-extends/scss/includes';

.breadcrumb {
    padding: 0;
    margin: 0;

    .active {
        font-weight: $font-weight-bold;
        margin-right: 0;
    }

    .breadcrumb-item {
        margin-right: 0.5rem;
        padding-left: 0;

       &::before {
            padding-right: 0.5rem;
        }
    }
}
</style>
