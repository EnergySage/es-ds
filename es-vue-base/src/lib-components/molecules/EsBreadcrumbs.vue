<template>
    <b-breadcrumb
        v-bind="$attrs"
        v-on="$listeners" />
</template>
<script>
import { BBreadcrumb } from 'bootstrap-vue';

export default {
    name: 'EsBreadcrumb',
    components: {
        BBreadcrumb,
    },
};
</script>

<style lang="scss" scoped>
@import '~@energysage/es-bs-base/scss/includes';

.breadcrumb {
    .active span {
        font-weight: $font-weight-bold;
    }
}
</style>
