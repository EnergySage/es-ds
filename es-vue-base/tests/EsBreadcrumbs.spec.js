import { mount } from '@vue/test-utils';
import EsBreadcrumbs from '@/src/lib-components/EsBreadcrumbs.vue';
import jestVue from '@/tests/jest.vue.config';

describe('EsBreadcrumb', () => {
    // Basic test; should exist for most components
    test('<EsBreadcrumb />', async () => {
        const wrapper = mount(EsBreadcrumbs, {
            ...jestVue,
        });
        const a11y = await axe(wrapper.element);

        expect(wrapper.vm).toBeTruthy();
        expect(wrapper.html()).toMatchSnapshot();
        expect(a11y).toHaveNoViolations();
    });

    test('check for <EsBreadcrumb />', async () => {
        const wrapper = mount(EsBreadcrumbs, {
            ...jestVue,
        });

        expect(wrapper.find('.breadcrumb').exists()).toBe(true);
    });
});
