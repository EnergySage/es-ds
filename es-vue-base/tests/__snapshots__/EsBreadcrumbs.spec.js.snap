// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EsBreadcrumb <EsBreadcrumb /> 1`] = `"<ol class="breadcrumb"></ol>"`;
